crates/metaslang/cst/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+pub mod cst;
+pub mod cursor;
+pub mod query;
+pub mod text_index;
+
+pub trait Kind:
+    Sized
+    + Copy
+    + Clone
+    + PartialEq
+    + Eq
+    + std::fmt::Display
+    + std::fmt::Debug
+    + serde::Serialize
+    + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
```

Comment:
Instead of relying on `strum` in our public API, and forcing users to use it, I wonder if it is possible to use the standard `std::str::FromStr` instead, and keep `strum` as an internal implementation detail.

Reply:
It has to be `TryFrom` because it is a partial function.

Reply:
`FromStr` also returns a `Result<>`:
https://doc.rust-lang.org/std/str/trait.FromStr.html

```rust
pub trait FromStr: Sized {
    type Err;

    fn from_str(s: &str) -> Result<Self, Self::Err>;
}
```

Reply:
and is also typically used for many other rust types via `"value".parse()`.

For us, we can use `#[derive(strum::EnumString)]` which auto-derives `std::str::FromStr` on the enum, but that is an implementation detail.

Reply:
the problem is that the `Err` associated type in `FromStr` need to implement `std::fmt::Debug`, but it isn't possible to constrain the associated type using stable Rust.

Reply:
Either you or @Xanewok may have a solution - the only one I have propagates the constraint on the associated type all through the codebase.

Reply:
I see. What do you think about simply using `String` as the error type for `FromStr`? this also has the advantage of providing an actual error message, instead of the (only) Unit variant `strum::ParseError::VariantNotFound`. The API will then be idiomatic, without exposing/depending on the highly-specific `strum`.

Not blocking though if it is not feasible. Just a minor suggestion.

Reply:
I tried that, but no joy, because the specification of the `Err` type still causes difficulties. My preference is to use an associated type constraint, as soon as it is stabilised. It is a significant improvement to the type system.

Reply:
I'll take quick look if we can simplify this.